optimum/configuration_utils.py

```diff
@@ -205,6 +218,15 @@ def _get_config_dict(
         from_auto_class = kwargs.pop("_from_auto", False)
         commit_hash = kwargs.pop("_commit_hash", None)
 
+        if use_auth_token is not None:
+            warnings.warn(
```

Comment:
```suggestion
            logger.warning(
```

Reply:
Interestingly Transformers also uses warnings.warn, not sure why the mix?

Reply:
my guess is that this is specific for FutureWarning, should we use logger everywhere ?

Reply:
I asked in https://huggingface.slack.com/archives/C01N44FJDHT/p1719313821448529
the answer is in https://stackoverflow.com/a/14762106 which refers to `logging` documentation

> warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning
> logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted